brachyura/src/main/java/io/github/coolcrabs/brachyura/ide/Intellijank.java

```diff
@@ -49,7 +49,27 @@ public void updateProject(Path projectDir, IdeProject ideProject) {
     public void updateProject(Path projectDir, IdeProject ideProject, @Nullable BaseJavaProject buildscript) {
         try {
             Path ideaPath = projectDir.resolve(".idea");
-            if (Files.exists(ideaPath)) PathUtil.deleteDirectory(ideaPath);
+
+            if (Files.exists(ideaPath)) {
+                // Delete unimportant files from .idea
+                Files.list(ideaPath).forEach(path -> {
+                    if (Files.isDirectory(path)) {
+                        // delete subdirectories
+                        PathUtil.deleteDirectory(path);
+                    } else if ("vcs.xml".equals(path.getFileName().toString())) {
+                        // do not delete vcs.xml, as it is important
+                        // otherwise we'll reset people's Git configurations
+                    } else {
+                        // delete other files
+                        try {
+                            Files.delete(path);
+                        } catch (IOException e) {
+                            Util.sneak(e);
+                        }
+                    }
+                });
```

Comment:
If you're deleting everything but `vcs.xml` I think you could shorten this to
```java
try (DirectoryStream<Path> files = Files.newDirectoryStream(ideaPath,
		file -> !"vcs.xml".equals(file.getFileName().toString()))) {
	for (Path file : files) {
		if (Files.isDirectory(file)) {
			PathUtil.deleteDirectory(file);
		} else {
			Files.delete(file);
		}
	}
}
```

Reply:
Thanks for the suggestion, will test this out later.